traffic_ops/app/db/reencrypt/reencrypt.go

```diff
@@ -0,0 +1,406 @@
+		os.Exit(0)
+	}
+
+	if err = updateKeyFile(previousKey, *previousKeyLocation, newKey); err != nil {
+		fmt.Println("updating the key file: ", err.Error())
+		os.Exit(0)
+	}
+
+	fmt.Println("Successfully re-encrypted keys for SSL Keys, URL Sig Keys, URI Signing Keys, and DNSSEC Keys.")
+}
+
+type Config struct {
+	DBName   string `json:"dbname"`
+	Hostname string `json:"hostname"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	Port     int    `json:"port"`
+	SSL      bool   `json:"ssl"`
+}
+
+func readKey(keyLocation string) ([]byte, error) {
+	var keyBase64 string
+	keyBase64Bytes, err := ioutil.ReadFile(keyLocation)
+	if err != nil {
+		return []byte{}, errors.New("reading file '" + keyLocation + "':" + err.Error())
+	}
+	keyBase64 = string(keyBase64Bytes)
+
+	key, err := base64.StdEncoding.DecodeString(keyBase64)
+	if err != nil {
+		return []byte{}, errors.New("AES key cannot be decoded from base64")
+	}
+
+	// verify the key works
+	_, err = aes.NewCipher(key)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return key, nil
+}
+
+func reEncryptSslKeys(db *sqlx.DB, previousKey []byte, newKey []byte) error {
+	tx, err := db.Begin()
+	if err != nil {
+		fmt.Println("transaction begin failed ", err, " ", tx)
+		os.Exit(0)
+	}
+	defer tx.Commit()
+
+	rows, err := tx.Query("SELECT id, data FROM sslkey")
+	if err != nil {
+		fmt.Println("querying: ", err)
+		os.Exit(0)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		updateTx, err := db.Begin()
+		if err != nil {
+			fmt.Println("transaction begin failed ", err, " ", updateTx)
+			os.Exit(0)
+		}
+		defer updateTx.Commit()
+
+		id := 0
+		var encryptedSslKeys []byte
+		if err = rows.Scan(&id, &encryptedSslKeys); err != nil {
+			fmt.Println("getting SSL Keys: ", err)
+			return err
+		}
+		jsonKeys, err := postgres.AesDecrypt(encryptedSslKeys, previousKey)
+		if err != nil {
+			fmt.Println("reading SSL Keys: ", err)
+			return err
+		}
+
+		reencryptedKeys, err := postgres.AesEncrypt(jsonKeys, newKey)
+		if err != nil {
+			fmt.Println("encrypting SSL Keys with new key: ", err)
+			return err
+		}
+
+		res, err := updateTx.Exec(`UPDATE sslkey SET data = $1 WHERE id = $2`, []byte(reencryptedKeys), id)
+		if err != nil {
+			fmt.Println("updating SSL Keys for id ", id, ": ", err)
+			return err
+		}
+		rowsAffected, err := res.RowsAffected()
+		if err != nil {
+			fmt.Println("error determining rows affected for reencrypting SSL Keys with id ", id)
+			return errors.New(fmt.Sprintf("determining rows affected for reencrypting SSL Keys with id %d", id))
+		}
+		if rowsAffected == 0 {
+			fmt.Println("no rows updated for reencrypting SSL Keys for id ", id)
+			return errors.New(fmt.Sprintf("no rows updated for reencrypting SSL Keys for id %d", id))
+		}
+
+	}
+
+	return nil
+}
+
+func reEncryptUrlSigKeys(db *sqlx.DB, previousKey []byte, newKey []byte) error {
+	tx, err := db.Begin()
+	if err != nil {
+		fmt.Println("transaction begin failed ", err, " ", tx)
+		os.Exit(0)
```

Comment:
Same as above RE: exit with failure code on failure and also returning error vs forcibly quitting.